### DISCUSSION TITLE
Repairing corrupt files

### DISCUSSION THREAD

**Discussion by OmikeOo** [gopro/labs#1000] [General]
I have a  Hero 12 and am using SD cards in the 'approved' list. I sucesfully recorded a number of videos without any issues but recently have a number which wont play and I get a corrupt file message on the PC. Have tried some of the suggestions I've seen on Youtube using dummy files etc but to no avail  (my GP does not seem to repair files?) Does anyone have a way to repair these files?

**Comment by dnewman-gpsw** [comment]
The only corruption a GoPro can have and can repair, is a truncated file. Just boot your GoPro and truncated file are automatically repaired. If you have other corruptions contact support. The Labs forum is not GoPro support.

You can manual force a file repair if the source camera was damaged, by following "Useful Tip - Manual File Recovery" section at the bottom https://gopro.github.io/labs/control/basename/

**Comment by ellielo** [comment]
Were you able to repair them? If the dummy file method and other suggestions haven’t worked, you might want to try a more reliable tool for repairing the corrupted GoPro files. I’ve had success using this tool: [https://news.macgasm.net/tips/fix-corrupted-gopro-video/](https://news.macgasm.net/tips/fix-corrupted-gopro-video/). It’s designed specifically for fixing corrupted GoPro videos and could be just what you need to recover your footage. Hopefully, it will work for you and save those files! Let me know if you need any help with the process

**Comment by OmikeOo** [comment]
No!

**Reply by yardenmultimediaay-dev** [reply]
Following up on dnewman's point — if in-camera SOS didn't fire and the dummy-file method also failed, the dummy probably didn't match the broken file's exact settings. Resolution, fps, and Protune profile all have to match the broken recording, not just be from the same camera.

If a matched dummy still doesn't fix it, the corruption is past the moov atom and needs a NAL-walker rather than a header patch. I run a free hosted service at ([repairmyvideo.com](https://repairmyvideo.com/)) that handles GoPro patterns specifically — worth a try before going to GoPro support.
